packages/flutter/lib/src/material/input.dart

```diff
@@ -526,6 +537,73 @@ class _InputState extends State<Input> {
 }
 
 /// A [FormField] that contains an [Input].
+///
+/// This is a convenience widget that simply wraps an [Input] widget in a
+/// [FormField]. The [FormField] maintains the current value of the [Input] so
+/// that you don't need to manage it yourself.
+///
+/// A [Form] ancestor is not required. The [Form] simply makes it easier to
+/// save, reset, or validate multiple fields at once. To use without a [Form],
+/// pass a [GlobalKey] to the constructor and use [GlobalKey.currentState] to
+/// save or reset the form field.
+///
+/// To see the use of [InputFormField], compare these two ways of a implementing
+/// a simple 2 text field form.
+///
+/// Using [InputFormField]:
+///
+/// ```dart
+/// String _firstName, _lastName;
+/// GlobalKey<FormState> _formKey = new GlobalKey<FormState>();
+/// ...
+/// new Form(
+///   key: _formKey,
+///   child: new Row(
+///     children: <Widget>[
+///       new InputFormField(
+///         labelText: 'First Name',
+///         onSaved: (InputValue val) { _firstName = val.text; }
+///       ),
+///       new InputFormField(
+///         labelText: 'Last Name',
+///         onSaved: (InputValue val) { _lastName = val.text; }
+///       ),
+///       new RaisedButton(
+///         child: new Text('SUBMIT'),
+///         // Note: Instead of _formKey.currentState, you could wrap the
+///         // RaisedButton in a Builder widget to get access to a BuildContext,
+///         // and use Form.of(context).
+///         onPressed: () { _formKey.currentState.save(); },
+///       ),
+///    )
+///  )
+/// ```
+///
+/// Using [Input] directly:
+///
+/// ```dart
+/// String _firstName, _lastName;
+/// InputValue _firstNameValue = const InputValue();
+/// InputValue _lastNameValue = const InputValue();
+/// ...
+/// new Row(
+///   children: <Widget>[
+///     new Input(
+///       value: _firstNameValue,
+///       labelText: 'First Name',
+///       onChanged: (InputValue val) { setState( () { _firstNameValue = val; } ); }
+///     ),
+///     new Input(
+///       value: _lastNameValue,
+///       labelText: 'Last Name',
+///       onChanged: (InputValue val) { setState( () { _lastNameValue = val; } ); }
+///     ),
+///     new RaisedButton(
+///       child: new Text('SUBMIT'),
+///       onPressed: () { _firstName = _firstNameValue.text; _lastName = _lastNameValue.text; },
```

Comment:
wrap this so it's longer, to make it obvious that there's more going on here than in the other example